subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/impl/DefaultFileSystemSnapshotter.java

```diff
@@ -182,15 +181,14 @@ public FileSystemSnapshotBuilder newFileSystemSnapshotBuilder() {
         return new FileSystemSnapshotBuilder(stringInterner, hasher);
     }
 
-    private FileSystemSnapshot filterSnapshot(FileSystemLocationSnapshot snapshot, PatternSet patterns) {
+    private FileSystemSnapshot filterSnapshot(FileSystemLocationSnapshot snapshot, SnapshottingFilter snapshottingFilter) {
         if (snapshot.getType() == FileType.Missing) {
             return FileSystemSnapshot.EMPTY;
         }
-        if (patterns.isEmpty()) {
+        if (snapshottingFilter.isEmpty()) {
```

Comment:
Nit: calling these parameters `filter` should be fine, too.

Reply:
Done.